core/src/main/scala/org/apache/gearpump/cluster/client/RunningApplication.scala

```diff
@@ -40,6 +44,16 @@ class RunningApplication(val appId: Int, master: ActorRef, timeout: Timeout) {
     }
   }
 
+  def waitUnilFinish(): Unit = {
+    val delegator = system.actorOf(Props(new AppResultListener(appId, master)))
+    val result = ActorUtil.askActor[ApplicationResult](delegator, WaitUntilFinish, INF_TIMEOUT)
+    result match {
+      case failed: ApplicationFailed =>
+        throw failed.error
+      case _ =>
+    }
+  }
+
   def askAppMaster[T](msg: Any): Future[T] = {
```

Comment:
The ActorUtil's askAppMaster will first `ResolveAppId` which is unnecessary here.